Added ability to add new keys

paolodona opened the pull request:
IYE is a fantastic little tool, the only missing thing for me was the ability to add new keys from the interface, as I find it a bit cumbersome to edit yml files with my text editor.

This PR adds this functionality:

![screenshot 2015-04-14 22 19 15](https://cloud.githubusercontent.com/assets/3245/7147655/50cb6c1e-e2f4-11e4-8814-fc7662423007.png)

It's quite simple but seems to work well enough for my basic requirements. I've decided to add a dropdown to choose the file you want to save the translation to, as people might be partitioning translations in different files.

Here is a video that shows the feature in action:  https://www.dropbox.com/s/qwyse1ymulprr5p/iye_new_keys.mov?dl=0

I hope you like it and merge it into the master branch.
Paolo

paolodona committed:
Added ability to add new keys

jacopomurador commented:
:+1: very cool!

paolodona committed:
create all locales in one go when creating new key

paolodona committed:
formatted new key form

vangberg commented:
This is great - thank you so much.

1. Make sure tests pass (`bundle && bundle exec rake`)
2. Let us make a separate page for the new form, instead of toggling with JS (eg. `/new`)
3. In our main Rails app we have 260 different locale files. It would be cool if we could use some kind of smart dropdown with search-as-you-type!
4. What do you think about allowing people to add new locales in the same form? Just have a button that, when clicked, adds a new input field/text area under the existing locales.

paolodona committed:
fixed tests, moved new to separate page, populated key on new action

paolodona committed:
add link to add new translations if none found

paolodona committed:
Merge branch 'master' of https://github.com/firmafon/iye

Conflicts:
	iye.gemspec

paolodona committed:
fixed gemspec

paolodona commented:
Hi Harry, a few updates:

- I've moved the new key form to a /new page
- I've added a message if a search returns no results with a link to create a new key
- In the file dropdown, I'm de-duping file names so you can just select the file radix, and enter all translations at once.
- Specs now pass

I think this is a good enough step to be merged in and address any further improvement as a separate PR. 

See video here: https://www.dropbox.com/s/kwu8wbhbneegwet/iye_new_key.mov?dl=0

stephenson commented:
:+1:  looks great!

paolodona committed:
added ability to delete key

paolodona committed:
Merge pull request #1 from paolodona/delete_key

Ability to delete key

vangberg commented on lib/i18n_yaml_editor/web.rb:
```diff
@@ -41,7 +41,11 @@ def app
         if translations = req["translations"]
           translations.each {|name, text|
-            app.store.translations[name].text = text
+            if text == '__delete__'
```
Let's split this out to a separate endpoint that takes a param `key`

```ruby
on delete, "delete" do
  …
end
```

vangberg replied:
And what we're doing is deleting a key AND it's translations, not just the translation.

vangberg commented on views/layout.erb:
```diff
@@ -128,13 +146,48 @@
       </div>
     </div>
 
+    <div class="new" style="display:none;">
```
Is this used anymore?

vangberg commented on views/new.html.erb:
```diff
@@ -0,0 +1,28 @@
+      </tr>
+      <tr>
+        <td>Key</td>
+        <td><input name='key' type="text" size="60" value="<%= req.params['key']%>"/></td>
```
We wan't to sanitize this somehow. If I press 'New' and there's a filter, I get this form:

![2015-04-24 at 15 58](https://cloud.githubusercontent.com/assets/252/7327387/b87317f6-ea9a-11e4-96b2-68e4516d2200.png)

We don't want the special characters there.

zinkkrysty commented:
This is cool, I'd like this merged into `iye`, would save me great amounts of time. How far are you from finishing/merging this in?

1st8 commented:
I incorporated these changes in my PR #20, trying to fix the issues mentioned here.

kaspth commented:
@paolodona Your improvements are cool, and we'd love to have them.

Are you still up for making this happen? If not, @1st8 has expressed interest in taking this over if you want to him to do it :grin:

paolodona commented:
Sadly I find myself quite busy with my day job so realistically I will not be a good & active maintainer right now. Happy for someone else to take it over and incorporate my changes. Thanks guys!

kaspth commented:
Sweet! We'll move the discussion to #20. Thanks for the work :grin:
